exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +74,2825 @@ func TestStart(t *testing.T) {
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	cp, err := getFirstTradablePair(t, asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := g.UpdateOrderbook(context.Background(), cp, asset.Futures); err != nil {
+		t.Errorf("%s UpdateOrderbook() error %v", g.Name, err)
+	}
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Empty); err != nil {
+		t.Errorf("%s GetWithdrawalsHistory() error %v", g.Name, err)
+	}
+}
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	currencyPair := currency.NewPair(currency.BTC, currency.USDT)
+	_, err := g.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	cp, err := getFirstTradablePair(t, asset.DeliveryFutures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), cp, asset.DeliveryFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), currencyPair, asset.Options)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderSubmission = &order.Submit{
+		Exchange: g.Name,
+		Pair: currency.Pair{
+			Delimiter: currency.UnderscoreDelimiter,
+			Base:      currency.BTC,
+			Quote:     currency.USDT,
+		},
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		AssetType: asset.CrossMargin,
+	}
+	_, err := g.SubmitOrder(context.Background(), orderSubmission)
+	if err != nil {
+		t.Errorf("Order failed to be placed: %v", err)
+	}
+}
+
+func TestCancelExchangeOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          currency.NewPair(currency.LTC, currency.BTC),
+		AssetType:     asset.Spot,
+	}
+	err := g.CancelOrder(context.Background(), orderCancellation)
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
 
-	var withdrawFiatRequest = withdraw.Request{}
-	_, err := g.WithdrawFiatFundsToInternationalBank(context.Background(),
-		&withdrawFiatRequest)
-	if err != common.ErrFunctionNotSupported {
-		t.Errorf("Expected '%v', received: '%v'", common.ErrFunctionNotSupported, err)
+func TestCancelBatchOrders(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	currencyPair := currency.NewPair(currency.LTC, currency.BTC)
+	_, err := g.CancelBatchOrders(context.Background(), []order.Cancel{
```

Comment:
This panics now after it successfully cancels orders:
```
--- FAIL: TestCancelBatchOrders (0.75s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

gocryptotrader/exchanges/gateio/gateio_wrapper.go:1224 +0xf4f
FAIL	github.com/thrasher-corp/gocryptotrader/exchanges/gateio	2.626s
```

[PRIOR NIT REF](https://github.com/thrasher-corp/gocryptotrader/pull/1058/files#r1051701674)